flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphTranslator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.graph;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.dag.Transformation;
+
+import java.util.Collection;
+
+/**
+ * A {@code Translator} is responsible for translating a given {@link Transformation}
+ * to its runtime implementation depending on the execution mode it is being executed.
+ * Currently, the execution mode can be either
+ * {@link org.apache.flink.streaming.api.RuntimeExecutionMode#STREAMING STREAMING} or
+ * {@link org.apache.flink.streaming.api.RuntimeExecutionMode#BATCH BATCH}.
```

Comment:
I think we don't need to refer to the API facing enum here, it's enough to know that we translate either for batch or streaming. Because the enum technically also has `AUTOMATIC`.